lib/grape/jsonapi/entity/resource.rb

```diff
@@ -28,6 +28,30 @@ def included_exposure
               end
             end
           end
+
+          private
+
+          def avoid_recursive_relation_check(nested_class)
+            return false if nested_class.root_exposures.empty?
+            nesting_exposures = nested_class.root_exposures.select do |exposure|
+              exposure.class == Grape::Entity::Exposure::NestingExposure
+            end
+            return false if nesting_exposures.empty?
+            result = nesting_exposures.detect do |exposure|
+              recursive_exposure?(exposure)
+            end
+            result
+          end
+
+          def recursive_exposure?(exposure)
+            if exposure.try(:nested_exposures)
+              exposure.nested_exposures.detect do |nested_exposure|
+                recursive_exposure?(nested_exposure)
+              end
+            else
+              exposure.send(:options).dig(:using) == self
```

Comment:
I feel like it might be better to use a named method or just rename the `using` key. I know we're using the `using` key as an identifier of some sort but it's difficult to figure out what it's for. In `self.nest` it reads more like `return if we're using nesting` and I don't think that's the intention (at least I think not).